lib/animina/accounts/resources/photo.ex

```diff
@@ -175,4 +195,75 @@ defmodule Animina.Accounts.Photo do
       reference :user, on_delete: :delete
     end
   end
+
+  def create_optimized_photos(record) do
+    case check_if_image_magick_is_installed() do
+      {:ok, _} ->
+        resize_images_with_imagemagick(record)
+
+      {:error, _} ->
+        copy_images_directly(record)
+    end
+  end
+
+  defp copy_images_directly(record) do
+    for type <- [:thumbnail, :normal, :big] do
+      OptimizedPhoto.create(%{
+        image_url: record.filename,
+        type: type,
+        user_id: record.user_id,
+        photo_id: record.id
+      })
+    end
+  end
+
+  # TODO: Confirm if the dimensions are  okay for each type
+  defp resize_images_with_imagemagick(record) do
+    for type <- [
+          %{
+            width: 100,
+            height: 100,
+            type: :thumbnail
+          },
+          %{
+            width: 300,
+            height: 300,
+            type: :normal
+          },
+          %{
+            width: 800,
+            height: 800,
+            type: :big
+          }
+        ] do
+      OptimizedPhoto.create(%{
+        image_url: resize_image(record.filename, type.width, type.height),
```

Comment:
Do we need the height? If yes you have to either calculate it or use a tool to figure it out.